spec/reek/smells/nested_iterators_spec.rb

```diff
@@ -100,152 +141,104 @@ def super_call_with_block
 
   it 'handles the case where super receives a block and arguments' do
     src = <<-EOS
-      def super_call_with_block
-        super(foo) do |k|
-          nothing.each { |thing| item }
+      def alfa
+        super do |bravo|
+          bravo.each { |charlie| charlie }
         end
       end
     EOS
 
     expect(src).to reek_of(:NestedIterators)
   end
 
-  describe 'sniff / warnings' do
-    let(:detector) { build(:smell_detector, smell_type: :NestedIterators) }
-
-    it 'reports a sensible warning message' do
-      source = <<-EOS
-        def foo
-          bar do |bar|
-            baz {|baz| }
-          end
-        end
-      EOS
-      expect(source).to reek_of(:NestedIterators, message: 'contains iterators nested 2 deep')
-    end
-
-    it 'reports the name of the method and line of the deepest iterator' do
-      source = <<-EOS
-        def foo
-          bar do |bar|
-            baz {|baz| }
-          end
-        end
-      EOS
-      expect(source).to reek_of(:NestedIterators, lines: [3])
-    end
-
-    it 'reports all lines on which nested iterators occur' do
-      source = <<-EOS
-        def bad
-          @fred.each {|item| item.each {|part| @joe.send} }
-          @jim.each {|ting| ting.each {|piece| @hal.send} }
-        end
-      EOS
-
-      expect(source).to reek_of(:NestedIterators, lines: [2, 3])
```

Comment:
No, that one has two cases with different nesting, and hence two smells, whereas this one has two cases with the same nesting, hence one smell with two lines.